artifacts.go

```diff
@@ -504,6 +506,76 @@ func signExisting(c *cli.Context) error {
 	return nil
 }
 
+func getDevices(msg *dbus.Signal) error {
+	if strings.HasSuffix(msg.Name, "InterfacesAdded") {
+		for _, v := range msg.Body {
+			if data, ok := v.(dbus.ObjectPath); ok {
+				if strings.Contains(string(data), "block_device") {
+					// TODO:
+					fmt.Printf("dev: %v\n", string(data))
+				}
+			}
+		}
+	}
+	return nil
+}
+
+func modifyArtifact(c *cli.Context) error {
+	if c.NArg() == 0 {
+		return cli.NewExitError("Nothing specified, nothing will be modified. \n"+
+			"Maybe you wanted to say 'artifacts read <pathspec>'?", 1)
+	}
+
+	// start connection with dbus
+	conn, err := dbus.SystemBus()
+	if err != nil {
+		return cli.NewExitError("Failed to connect to dbus: "+err.Error(), 1)
+	}
+	defer conn.Close()
+
+	if !conn.SupportsUnixFDs() {
+		return cli.NewExitError("Connection does not support unix fsd; exiting", 1)
+	}
+
+	file, err := os.OpenFile(c.Args().First(), os.O_RDWR, 0)
```

Comment:
Maybe I'm a bit confused here, but this will not enable you to modify an artifact, will it? Because an artifact has the payload inside the tar structure, and here you're opening the file directly.

Reply:
Yes, there is a separate task for figuring out what kind of file you are reading. This is just an interface for d-bus and udisks.